azure/services/vmextensions/spec.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vmextensions
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-04-01/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-provider-azure/azure"
+)
+
+// VMExtensionSpec defines the specification for a VM or VMScaleSet extension.
+type VMExtensionSpec struct {
+	Spec          azure.ExtensionSpec
```

Comment:
This could just be:

```suggestion
	azure.ExtensionSpec
```

so you can refer to fields directly, for example `VMExtensionSpec.Publisher` instead of `VMExtensionSpec.Spec.Publisher`